### ISSUE TITLE
"Silent" mode?

### ISSUE THREAD

**Issue by mikaylathompson** [liviu-/ding#7]
I'd like to run these in the background, i.e. `./ding/ding.py in 60s &`, but the countdown printer means that it clears the screen and is constantly overwriting my prompt.

Would you accept a PR for a silent mode (perhaps with a less confusing name--I don't mean silencing the beep, just the print output), so it could be run as `./ding/ding.py in 60s --silent &` and only interrupt you once time was up?

**Comment by liviu-** [comment]
That's an interesting use case! I can't test it right now, but would `$ ding in 1s >> /dev/null &` work instead?

Edit: Wait, no, that would send the bell code to `/dev/null` as well.

**Comment by mikaylathompson** [comment]
Yeah, that captures the dings as well.  So that's the `very silent` mode. 😛

**Comment by liviu-** [comment]
Haha, yup.  I guess that would work if the beep was written to `stderr`, but then that wouldn't make much sense.

So I took a look at it, and it's definitely doable and a PR would be welcome. The argument parser that I made may be a bit annoying to work with, but I believe that a nice way to add arguments, and circumvent the `check_input()` checks without too much hassle is to add a constant `ARGS = ['--no-timer']`, and then the first line in `check_inputs()` could be something like this:

``` python
# Ignore optional args
args = sorted((set(args) - set(ARGS)), key=lambda x: args.index(x)) # preserve order
```

which would ensure that the rest of the function ignores the arguments from `ARGS`.

In the `main()` function you'd still know if the argument was passed via `'--no-timer' in args`. The part that I'm more concerned about is circumventing my bad design of having the timer itself in the `print_time()` function :stuck_out_tongue: It can be solved with a bunch of `if`s but it may be a bit ugly, and naming a function `print_time()` when it prints nothing is a bit confusing.

But yes, sure, if you don't mind crafting a PR, I'd be glad to merge it. Feel free to use different implementation that the one I suggested if you can find something that works better. If you don't have time, let me know and I'll push a patch myself this weekend.

**Comment by simison** [comment]
This is exactly how I'd like to use this!

**Comment by mikaylathompson** [comment]
Just sent a PR (#9), I went with assuming that `--no-timer` had to be the final arg, and popping it off before sending it to `check_input`, so it doesn't mess up anything there.

Renamed `print_time` to `countdown`, and I agree that all the `if`s are somewhat ugly, but hopefully not too terrible.  Definitely feel free to rework that if you have something better.

**Comment by liviu-** [comment]
The PR was merged and was released in 2.0.0 (`$ pip install --upgrade ding-ding` if installed via `pip`).

Many thanks for the help!
